### PR TITLE
THIRDPARTY.md cleanup

### PR DESCRIPTION
**Pull request by SamuraiAku**
This purpose of this PR is to review THIRDPARTY.md and determine where it needs to be updated. 

During the review of of #41095 a few instances of out-of-date information were determined, but this list was not exhaustive. This PR will review all of the entries in THIRDPARTY.md and validate each one. This PR will not attempt major reformatting of THIRDPARTY or refactoring of the third party code to make it more obvious. That exercise will be left for future PRs.

Most of the changes are clearly explained in the commit log and don't require repetition here. Below are listed items where I would like reviews from the core dev team.

**Third Party Code in src**    
- [LLVM]
    + src/jitlayers.c - The file header says that portions of the code are copied from the LLVM project and marked below. But in searching the file, I could not find a single such marking. In the absence of such markings, I chose to remove references to jitlayers.c from THIRDPARTY.md.    
- [Python]
    + Removed reference to joinpath from THIRDPARTY.md because I could not find a function with that name. 
- [Google Benchmark]
    + In #41095 @vtjnash said that cyclecount should not have been listed at all.  I can't find any evidence that cyclecount exists in the codebase, so I removed references from THIRDPARTY.md

**Third Party Code in Base**   
- base/grisu/*    
    + Deleted reference per comment from vtjnash in #41095

**Standard Library Dependencies**
- [FEMTOLISP] 
    + I feel that FEMTOLISP does not belong in the dependencies section of THIRDPARTY.md.  Everything else in this section exists in the deps/ directory and pulls its code from an outside repository. But the femtolisp code exists in src/flisp.  Since the code was first added to Julia in the early days, it has gone substantial revision within this repository. At the same time @JeffBezanson has been making updates to the code in the original repository in parallel with Julia.  A quick diff indicates that the two code bases are not identical. As a result, I have moved the femtolisp reference to the third party code in src/ section of the document.  I have also added the BSD-3 license to the src/flisp directory.

- I found three dependencies in deps/ that were not listed in THIRDPARTY.md.  I added all three to the list of stdlib dependencies. Can the dev team comment if this is the appropriate location?

### CODE REVIEW

File: THIRDPARTY.md [hunk 1]
```diff
@@ -1,46 +1,47 @@
...
-- [MBEDTLS](https://tls.mbed.org/how-to-get) [either GPLv2 or Apache 2.0]
+- [MBEDTLS](https://github.com/ARMmbed/mbedtls/blob/development/LICENSE) [Apache 2.0]
 - [MPFR](https://www.mpfr.org/mpfr-current/mpfr.html#Copying) [LGPL3+]
 - [OPENBLAS](https://raw.github.com/xianyi/OpenBLAS/master/LICENSE) [BSD-3]
 - [LAPACK](https://netlib.org/lapack/LICENSE.txt) [BSD-3]
 - [PCRE](https://www.pcre.org/licence.txt) [BSD-3]
-- [SUITESPARSE](http://suitesparse.com) [mix of LGPL2+ and GPL2+; see individual module licenses]
+- [SUITESPARSE](https://github.com/DrTimothyAldenDavis/SuiteSparse/blob/master/LICENSE.txt) [mix of LGPL2+ and GPL2+; see individual module licenses]
+- [LIBBLASTRAMPOLINE](https://github.com/staticfloat/libblastrampoline/blob/main/LICENSE) [MIT]
+- [LIBWHICH](https://github.com/vtjnash/libwhich/blob/master/LICENSE) [MIT]
```
Comment (vtjnash) [inline THIRDPARTY.md:43]: libwhich should be in the next list (build process tools)

Comment (SamuraiAku) [reply THIRDPARTY.md]: Moved.

**Comment by DilumAluthge** [comment]
@JeffBezanson can probably weight in on the femtolisp stuff.

**Comment by ViralBShah** [comment]
@SamuraiAku Thank you for the meticulous work on cleaning this up.

**Comment by ViralBShah** [comment]
I've merged this, since it is a major improvement. Can always fine tune going forward.

**Pull request merged**